improver_tests/calibration/ensemble_calibration/test_ApplyCoefficientsForEnsembleCalibration.py

```diff
@@ -475,36 +408,25 @@ def test_end_to_end(self):
     @ManageWarnings(
         ignored_messages=["Collapsing a non-contiguous coordinate."])
     def test_end_to_end_with_mask(self):
-        """An example end-to-end calculation, this time using a mask to limit
-        the application of calibration to regions where the mask=1."""
+        """An example end-to-end calculation, but making sure that the
+        areas that are masked within the landsea mask, are masked at the
+        end."""
 
         # Construct a mask and encapsulate as a cube.
-        mask = np.ones((3, 3))
-        mask[1:, 1:] = 0
+        mask = np.array([[1, 0, 0], [0, 1, 0], [0, 0, 1]])
         mask_cube = self.current_temperature_forecast_cube[0].copy(data=mask)
-
-        # Collapse input realizations to get uncalibrated expectation values.
-        forecast_predictors = self.current_temperature_forecast_cube.collapsed(
-            "realization", iris.analysis.MEAN)
-        forecast_vars = self.current_temperature_forecast_cube.collapsed(
-            "realization", iris.analysis.VARIANCE)
-
-        # Manually construct merged calibrated and uncalibrated arrays.
-        self.expected_loc_param_mean[1:, 1:] = (
-            forecast_predictors.data[1:, 1:])
-        self.expected_scale_param_mean[1:, 1:] = (
-            forecast_vars.data[1:, 1:])
+        mask_loc_param = np.ma.masked_where(mask, self.expected_loc_param_mean)
```

Comment:
I think in stage data, the land-sea mask has land points as 1 and sea points as 0, so this might be doing the opposite of what you want. I think we probably need to put the information about what form the land-sea mask takes in the docstring to avoid confusion.

Reply:
I have added a doc string to address this.

Reply:
I have also flipped the mask (which hopefully addresses this)